app/javascript/controllers/caption_controller.js

```diff
@@ -12,11 +12,32 @@ export default class extends Controller {
     this.element.querySelector('form').reset()
   }
 
+  ocrAvailable () {
+    return this.ocrContentTypes().indexOf(this.contentStructureTarget.value) >= 0 && this.ocrEnabled()
+  }
+
+  sttAvailable () {
+    return this.sttContentTypes().indexOf(this.contentStructureTarget.value) >= 0 && this.sttEnabled()
+  }
+
   // list of content structures that are allowed to run OCR
-  ocrAllowed () {
+  ocrContentTypes () {
     return ['simple_image', 'simple_book', 'document']
   }
 
+  // list of content structures that are allowed to run speech to text
+  sttContentTypes () {
+    return ['media']
+  }
+
+  ocrEnabled () {
+    return this.data.get('ocr-enabled') === 'true'
+  }
+
+  sttEnabled () {
+    return this.data.get('stt-enabled') === 'true'
+  }
```

Comment:
these enable to see the `Settings`, which are passed into the stimulus controller